### PR TITLE
[Indexer gRPC] Data Service Live Mode

### PR DESCRIPTION
**Pull request by just-in-chang**
## Description
For the data service, Instead of returning an error when not specifying a starting version, we just start from the latest transaction (live mode). We get the latest version from the in-memory cache. 

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [x] Other (specify)

## How Has This Been Tested?

image

Data is live!

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by larry-aptos — APPROVED** [review]

File: ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs [hunk 1]
```diff
@@ -152,9 +152,8 @@ impl RawData for RawDataServerWrapper {
         let (tx, rx) = channel(self.data_service_response_channel_size);
         let current_version = match &request.starting_version {
             Some(version) => *version,
-            None => {
-                return Result::Err(Status::aborted("Starting version is not set"));
-            },
+            // Live mode if starting version isn't specified
+            None => (self.in_memory_cache.latest_version().await - 1).max(0), // Smallest version is 0 (genesis)
```
Comment (CapCap) [inline ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs:156]: I don't think the max is necessary; latest_version returns a u64, so negative numbers are impossible

Comment (larry-aptos) [inline ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs:156]: saturating_sub? just in case cache is not init'ed yet.

**Review by larry-aptos — APPROVED** [review]

File: ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs [hunk 2]
```diff
@@ -152,9 +152,12 @@ impl RawData for RawDataServerWrapper {
...
             Some(version) => *version,
-            None => {
-                return Result::Err(Status::aborted("Starting version is not set"));
-            },
+            // Live mode if starting version isn't specified
+            None => self
+                .in_memory_cache
+                .latest_version()
+                .await
+                .saturating_sub(1),
```
Comment (CapCap) [inline ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs:160]: TIL

**Review by CapCap — APPROVED** [review]

**Review by rtso — APPROVED** [review]
nice!

**Pull request merged**